base/inference.jl

```diff
@@ -3478,6 +3396,17 @@ function reindex_labels!(linfo::LambdaInfo, sv::InferenceState)
     end
 end
 
+@eval function return_type(f::ANY, t::ANY)
+    $(Expr(:meta, :pure))
```

Comment:
I still don't really get this. Even after we've inferred `Type{Int}` for `return_type(identity, Tuple{Int})`, we still have to actually call `return_type` at run time? That seems at best unnecessary, and at worst creates an opportunity for the run-time `return_type` to disagree.